src/megaapi_impl.cpp

```diff
@@ -8876,16 +8860,16 @@ void MegaApiImpl::setExcludedRegularExpressions(MegaSync *sync, MegaRegExp *regE
         return;
     }
 
-    int tag = sync->getTag();
-    sdkMutex.lock();
-    if (syncMap.find(tag) == syncMap.end())
-    {
-        sdkMutex.unlock();
-        return;
-    }
-    MegaSyncPrivate* megaSync = syncMap.at(tag);
-    megaSync->setRegExp(regExp);
-    sdkMutex.unlock();
+    SdkMutexGuard g(sdkMutex);
+    client->syncs.forEachUnifiedSync([&](UnifiedSync& s) {
+        if (s.mConfig.getTag() == sync->getTag())
+        {
+            s.mConfig.setRegExps(regExpToVector(regExp));
+            client->syncs.saveSyncConfig(s.mConfig);
```

Comment:
:raised_eyebrow:  Did we have a bug here? The update of the regular expression seems to not be saved in the DB previously. Good that they are only used globally, not per sync.

Reply:
that's a good point about it those being a global setting, I also was thinking it may have been a bug.  However, the SyncConfig serialization is storing all the regExp strings.   Does this mean we have a mismatch between design and implementation?   Or, are we transitioning to per-Sync settings for the exclusion rules?

Reply:
Actually, we do (were?) planning to provide exclusion rules per sync at some point, rather than globally. It's a popular user's demand. It's in line with the `.megaignore` (that we should finalize at some point) :smiley:

Reply:
Ok, sounds good to me :-)